### PR TITLE
Use group ID instead of group name

### PR DESCRIPTION
**Pull request by lcvisser**
MacOS allows non-standard group names, including spaces and backslashes, which breaks building from source. This change circumvents the issue by using group ID (numerical) instead of group names.

Note that this fixes only building macports-base, not individual ports.

Related ticket: https://trac.macports.org/ticket/49501

This changes is based on the contribution of user chucko58 (Chuck Fry) in this comment: https://trac.macports.org/ticket/49501#comment:7

### CODE REVIEW

**Comment by lcvisser** [comment]
Should I do something to here?

**Comment by raimue** [comment]
Hey! Thank you for the contribution. I guess this hangs because it is a niche problem which is harder to test than usual.

The change seems reasonable to me without any testing. But shouldn't we go with the numeric ID for both user and group for consistency?

File: doc/prefix.mtree.in [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # MacPorts filesystem hierarchy, for internal use only. Changes to this file will not stick across installations.
 
-/set type=dir uname=@DSTUSR@ gname=@DSTGRP@ mode=@DSTMODE@
+/set type=dir uname=@DSTUSR@ gid=@DSTGID@ mode=@DSTMODE@
```
Comment (jmroot) [inline doc/prefix.mtree.in:3]: This is OK for local installation, but the mtree files also get shipped in the binary packages. I don't think we can guarantee that the gid will refer to the same user on the destination machine as on the build machine.

**Review by jmroot — COMMENTED** [review]
Maybe a configuration option to refer to the group numerically would be more appropriate? Or if there's a way to make mtree handle the exotic group names correctly, that would be best of all.

**Comment by jmroot** [comment]
Or, since this only seems to be an issue for non-root installations, would it work to remove the uid and git options entirely in that case?

File: doc/prefix.mtree.in (hunk 1 above)
Comment (lcvisser) [reply doc/prefix.mtree.in:3]: How does that work then with the group name? That has the same problem right? Or am I missing something?

**Comment by lcvisser** [comment]
> Or, since this only seems to be an issue for non-root installations, would it work to remove the uname and gname options entirely in that case?

Is that possible?

Or can we have options like `--use-gid=xyz` to `configure` to override? I'm a bit too far out of my comfort zone here to see how to do that, but I can put in some effort, considering it's so niche...

 My organisation should fix the group names (it's something like `myorgAD\Domain Users`: a `\` and a space), but since that is not going to happen, I'm stuck with it. Currently I need to build macports-base from source every time there's an OS update, which is not the end of the world.

File: doc/prefix.mtree.in (hunk 1 above)
Comment (jmroot) [reply doc/prefix.mtree.in:3]: The user and group are recorded by name in the pkg.

**Comment by jmroot** [comment]
> Is that possible?

Anything's possible, it's just a text file. :)

File: doc/prefix.mtree.in (hunk 1 above)
Comment (lcvisser) [reply doc/prefix.mtree.in:3]: But the group name might be invalid on the destination machine too, or am I just completely misunderstanding this?

Comment (jmroot) [reply doc/prefix.mtree.in:3]: We can be pretty sure there will be a group named `admin` on macOS systems.

**Pull request closed without merging**